install/operator/pkg/generator/generator_test.go

```diff
@@ -6,15 +6,44 @@ import (
 	"strings"
 	"testing"
 
+	osappsv1 "github.com/openshift/api/apps/v1"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/syndesisio/syndesis/install/operator/pkg/apis/syndesis/v1beta1"
 	"github.com/syndesisio/syndesis/install/operator/pkg/generator"
 	"github.com/syndesisio/syndesis/install/operator/pkg/syndesis/configuration"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
+//
+// Registers the DeploymentConfig type and adds in a
+// mock syndesis-db deployment config runtime object
+//
+func fakeClient() client.Client {
```

Comment:
This has to be added so that the fakeclient knows about v1.DeploymentConfig. This is not so surprising given this client is a basic fake and not aware of openshift APIs.